sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_models.hpp

```diff
@@ -0,0 +1,772 @@
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction DecipherOnly;
+  };
+
+  /**
+   * @brief An action to be executed at a prescribed time in a certificates lifecycle.
+   *
+   */
+  struct LifetimeAction final
+  {
+    /**
+     * @brief Get the CertificatePolicyAction to be performed.
+     *
+     */
+    CertificatePolicyAction Action;
+
+    /**
+     * @brief Get the action should be performed the specified number of days before the certificate
+     * will expire.
+     *
+     */
+    Azure::Nullable<int32_t> DaysBeforeExpiry;
+
+    /**
+     * @brief Get the action should be performed when the certificate reaches the specified
+     * percentage of its lifetime. Valid values include 1-99.
+     *
+     */
+    Azure::Nullable<int32_t> LifetimePercentage;
+  };
+
+  /**
+   * @brief Properties of a CertificateIssuer.
+   *
+   */
+  struct IssuerParameters final
+  {
+    /**
+     * @brief Indicates if the certificates generated under this policy should be published to
+     * certificate transparency logs.
+     *
+     */
+    Azure::Nullable<bool> CertTransparency;
+
+    /**
+     * @brief Certificate type as supported by the provider (optional); for example 'OV-SSL',
+     * 'EV-SSL'.
+     *
+     */
+    Azure::Nullable<std::string> Cty;
+
+    /**
+     * @brief Name of the referenced issuer object or reserved names; for example, 'Self' or
+     'Unknown'.
+     *
+     */
+    Azure::Nullable<std::string> Name;
+  };
+
+  /**
+   * @brief A policy which governs the lifecycle a properties of a certificate managed by Azure Key
+   * Vault.
+   *
+   */
+  struct CertificatePolicy final
+  {
+    /**
+     * @brief Get the type of backing key to be generated when issuing new certificates.
+     *
+     */
+    Azure::Nullable<CertificateKeyType> KeyType;
+
+    /**
+     * @brief Get a value indicating whether the certificate key should be reused when rotating the
+     * certificate.
+     *
+     */
+    Azure::Nullable<bool> ReuseKey;
+
+    /**
+     * @brief Get a value indicating whether the certificate key is exportable from the vault or
+     * secure certificate store.
+     *
+     */
+    Azure::Nullable<bool> Exportable;
+
+    /**
+     * @brief Get the curve which back an Elliptic Curve (EC) key.
+     *
+     */
+    Azure::Nullable<CertificateKeyCurveName> KeyCurveName;
+
+    /**
+     * @brief Get the size of the RSA key. The value must be a valid RSA key length such as 2048 or
+     * 4092.
+     *
+     */
+    Azure::Nullable<int32_t> KeySize;
+
+    /**
+     * @brief Get the subject name of a certificate.
+     *
+     */
+    std::string Subject;
+
+    /**
+     * @brief Get the subject alternative names (SANs) of a certificate.
+     *
+     */
+    SubjectAlternativeNameList SubjectAlternativeNames;
+
+    /**
+     * @brief Get the issuer for a certificate.
+     *
+     */
+    IssuerParameters Issuer;
```

Comment:
We only expose the `IssuerName` and the rest of `IssuerParameters` can be hidden.